### PR TITLE
feat: add ethereum address mapping

### PR DESCRIPTION
**Pull request by meelrossi**
Add ethereum address mapping for decentraland names

image
image

### CODE REVIEW

File: src/modules/ens/sagas.ts [hunk 1]
```diff
@@ -304,6 +309,25 @@ export function* ensSaga(builderClient: BuilderClient, ensApi: ENSApi) {
...
+  function* handleSetENSAddressRequest(action: SetENSAddressRequestAction) {
+    const { ens, address } = action.payload
+    try {
+      const wallet: Wallet = yield getWallet()
+      const signer: ethers.Signer = yield getSigner()
+
+      const nodehash = namehash(ens.subdomain)
+      const resolverContract = ENSResolver__factory.connect(ENS_RESOLVER_ADDRESS, signer)
+
+      const transaction: ethers.ContractTransaction = yield call(() => resolverContract['setAddr(bytes32,address)'](nodehash, address))
```
Comment (LautaroPetaccio) [inline src/modules/ens/sagas.ts:321]: There are lots of parts if the Builder where `call` is wrongly used 😢 . 
```suggestion
      const transaction: ethers.ContractTransaction = yield call([resolverContract, 'setAddr(bytes32,address)'], nodehash, address)
```

File: src/modules/ens/sagas.ts [hunk 2]
```diff
@@ -304,6 +309,25 @@ export function* ensSaga(builderClient: BuilderClient, ensApi: ENSApi) {
     }
   }
 
+  function* handleSetENSAddressRequest(action: SetENSAddressRequestAction) {
+    const { ens, address } = action.payload
```
Comment (LautaroPetaccio) [inline src/modules/ens/sagas.ts:313]: Would you mind adding tests for this handler?

File: src/modules/ens/reducer.ts [hunk 3]
```diff
@@ -285,6 +296,21 @@ export function ensReducer(state: ENSState = INITIAL_STATE, action: ENSReducerAc
             }
           }
         }
+        case SET_ENS_ADDRESS_SUCCESS: {
+          const { ens, address } = transaction.payload
```
Comment (LautaroPetaccio) [inline src/modules/ens/reducer.ts:300]: Would you mind adding tests for this reducer?

File: src/components/Modals/ENSMapAddressModal/ENSMapAddressModal.tsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  return (
+    <Modal closeIcon={<Close />} onClose={onClose} size="tiny">
+      <div className={styles.main}>
+        <div className={styles.info}>
+          <h1 className={styles.title}>{t('ens_map_address_modal.title')}</h1>
+          <span className={styles.description}>{t('ens_map_address_modal.description')}</span>
+        </div>
+        <div>
+          <Field
+            id="address-input"
```
Comment (LautaroPetaccio) [inline src/components/Modals/ENSMapAddressModal/ENSMapAddressModal.tsx:37]: What do you think about adding the type `address` to the Field component?
```suggestion
            id="address-input"
            type="address"
```

File: src/components/Modals/ENSMapAddressModal/ENSMapAddressModal.tsx [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            label={t('ens_map_address_modal.address.label')}
+            value={address}
+            placeholder={t('ens_map_address_modal.address.placeholder')}
+            onChange={handleAddressChange}
```
Comment (LautaroPetaccio) [inline src/components/Modals/ENSMapAddressModal/ENSMapAddressModal.tsx:41]: What do you think about disabled the field when loading?

File: src/components/Modals/ENSMapAddressModal/ENSMapAddressModal.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import { useCallback, useEffect, useState } from 'react'
+import { Button, Close, Field, Message } from 'decentraland-ui'
+import Modal from 'decentraland-dapps/dist/containers/Modal'
+import ethereumImg from '../../../icons/ethereum.svg'
+import { Props } from './ENSMapAddressModal.types'
+import styles from './ENSMapAddressModal.module.css'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+
+export default function EnsMapAddressModal(props: Props) {
```
Comment (LautaroPetaccio) [inline src/components/Modals/ENSMapAddressModal/ENSMapAddressModal.tsx:9]: Would you mind adding some tests for this modal?

File: src/components/ENSListPage/ENSListPage.tsx [hunk 7]
```diff
@@ -238,8 +310,95 @@ export default class ENSListPage extends React.PureComponent<Props, State> {
...
+                      </span>
+                    ) : (
+                      <Button compact className="ens-list-btn" onClick={this.handleAssignENSAddress.bind(null, ens)}>
+                        <Icon name="add" />
+                        {t('ens_list_page.button.link_to_address')}
+                      </Button>
+                    )}
+                  </Table.Cell>
+                  <Table.Cell>{this.renderLandLinkInfo(ens)}</Table.Cell>
+                  <Table.Cell>TBD</Table.Cell>
```
Comment (LautaroPetaccio) [inline src/components/ENSListPage/ENSListPage.tsx:381]: Are we missing something here?

File: src/modules/translation/languages/en.json [hunk 8]
```diff
@@ -580,20 +580,41 @@
...
     "assigned_to_land": "Land ({landId})",
     "assigned_to_estate": "Estate ({landId})",
     "button": {
       "assign": "Use as Link",
       "edit": "Edit Link",
-      "use_as_alias": "Use as Alias"
+      "use_as_alias": "Use as Alias",
+      "add_to_avatar": "Add to Avatar",
+      "link_to_address": "Link to an Ethereum Address",
+      "assign_to_land": "Assign to Land o WORLDs"
```
Comment (LautaroPetaccio) [inline src/modules/translation/languages/en.json:598]: I think the name can be assigned to only one World, right?
```suggestion
      "assign_to_land": "Assign to Land or World"
```

File: src/components/Modals/ENSMapAddressModal/ENSMapAddressModal.tsx [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    },
+    [setAddress]
+  )
+
+  useEffect(() => {
+    document.getElementById('address-input')?.focus()
+  }, [])
+
+  return (
+    <Modal closeIcon={<Close />} onClose={onClose} size="tiny">
```
Comment (LautaroPetaccio) [inline src/components/Modals/ENSMapAddressModal/ENSMapAddressModal.tsx:29]: I think we're not preventing here the closing of the modal when the `isLoading` property is set, what do you think about adding it?

**Review by LautaroPetaccio — COMMENTED** [review]
Looks great! 🥇 
I've left some comments to review

File: src/components/ENSListPage/ENSListPage.tsx (hunk 7 above)
Comment (meelrossi) [reply src/components/ENSListPage/ENSListPage.tsx:381]: Yes! There is going to be a Name detail page and here there is an edit button. I will add it in another PR
image

File: src/modules/ens/reducer.spec.ts [hunk 10]
```diff
@@ -86,3 +97,83 @@ describe('when handling the fetch external names actions', () => {
...
+    })
+
+    it('should set the error as null', () => {
+      const txSuccessAction = fetchTransactionSuccess({
+        actionType: SET_ENS_ADDRESS_SUCCESS,
+        payload: { ens, address, chainId: ChainId.ETHEREUM_SEPOLIA, hash: 'hash' }
+      } as Transaction)
+      const newState = ensReducer(state, txSuccessAction)
+      expect(newState.error).toEqual(null)
+    })
```
Comment (LautaroPetaccio) [inline src/modules/ens/reducer.spec.ts:177]: It seems that adding the `ensAddressRecord` to the map is not checked here, would you mind adding it?

**Review by LautaroPetaccio — APPROVED** [review]
🚀

**Pull request merged**